doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+    as analysis
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config = CollectionConfig(
+        name='collection_name',
+        parameters={
+            "key": "value"
+        }
+    )
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission'
+        hook_parameters={
+            "key": "value"
+        },
+        collection_configs=[
+            collection_config,
+        ]
+    )
+
+    estimator = TensorFlow(
+        role=role
+        train_instance_count=1,
+        train_instance_type=train_instance_type,
+        debugger_hook_config=debugger_hook_config
+    )
+
+When you call ``fit()`` on the ``estimator``, SageMaker creates a training job for you and the debugging hook is initialized using the parameters provided to the ``DebuggerHookConfig`` object. This hook starts emitting the relevant debugging data i.e. the tensor collections in real time and stores the data locally in the local path provided in ``DebuggerHookConfig``. This data is then uploaded to an S3 path derived from the path provided in the hook configuration in near real time:
+
+.. code::
+
+    s3://{destination-bucket-prefix}/debug-output/
+
+To access the above S3 path through the estimator object, you can do:
+
+.. code:: python
+
+    tensors_s3_output_path = estimator.latest_job_debugger_artifacts_path()
+
+
+You can use the ``S3Downloader`` utility to view and download the debugging data emitted during training in Amazon S3 (note that data is stored in a streaming fashion so the data you download locally through ``S3Downloader`` will be a snapshot of the data generated until that time):
+
+.. code:: python
+
+    from sagemaker.s3 import S3Downloader
+
+    # Start the training by calling fit
+    # Setting the wait to `False` would make the fit asynchronous
+    estimator.fit(wait=False)
+
+    # Get a list of S3 URIs
+    S3Downloader.list('s3://path/for/data/emission/debug-output')
+
+    # Read a specific file
+    S3Downloader.read_file('s3://path/for/data/emission/debug-output/events/0000000000/event-filename.tf')
+
+The contents of the single TF-Event file should be all the data emitted for a particular step in the training process.
+
+Specifying individual configuration for each collection
+-------------------------------------------------------
+
+When planning to emit and store a collection, you have the option of specifying ``collection_parameters`` in the ``CollectionConfig`` object. The collection parameters define properties of storage like ``save_interval`` i.e the step frequency at which a particular collection is saved, ``include_regex`` i.e. the regex of the tensors to be stored etc. You can define these values for each of the ``CollectionConfig`` object you intend to store.
```

Comment:
Addressed, but not exactly as the suggestion. Changed the wording a tad bit since I explained the significance of the `parameters` separately earlier (as part of new revision).